### PR TITLE
Added functionality to improve failure triaging

### PR DESCRIPTION
**Pull request by adityapatwardhan**
- Nunit logs will be created to tests.
- Logs are copied to the Azure share.
- Pester is run with -Quiet
- Elevated and unelevated runs for tests with appropriate tags

### CODE REVIEW

File: test/tools/CodeCoverageAutomation/Start-CodeCoverageRun.ps1 [hunk 1]
```diff
@@ -99,19 +136,48 @@ try
...
 
-    $coverallsParams | % { Write-LogPassThru -Message $_ }
+    $coverallsParams | ForEach-Object { Write-LogPassThru -Message $_ }
 
+    Write-LogPassThru -Message "Uploading to CoverAlls"
     & $coverallsExe """$coverallsParams"""
 
+    $bashScriptInvoker = "$PSScriptRoot\CodecovUploader.cmd"
+    $bashScript = "$PSScriptRoot\CodecovUploader.sh"
+    $cygwinLocation = "$env:SystemDrive\cygwin*"
```
Comment (JamesWTruher) [inline test/tools/CodeCoverageAutomation/Start-CodeCoverageRun.ps1]: >cygwinLocation [](start = 5, length = 14)

not used?

Comment (adityapatwardhan) [inline test/tools/CodeCoverageAutomation/Start-CodeCoverageRun.ps1]: Fixed. Should have been used in the if condition.

File: test/tools/CodeCoverageAutomation/Start-CodeCoverageRun.ps1 [hunk 2]
```diff
@@ -99,19 +136,48 @@ try
...
+
+    if($bashScript)
+    {
+        Remove-Item $bashScript -Force
+    }
+
+    Invoke-RestMethod 'https://codecov.io/bash' -OutFile $bashScript
+    Write-BashInvokerScript -path $bashScriptInvoker
+
+    if(Test-Path $bashScriptInvoker -and $bashScript -and $cygwinLocation)
```
Comment (JamesWTruher) [inline test/tools/CodeCoverageAutomation/Start-CodeCoverageRun.ps1]: >t-Path $bashScriptInvoker -and $bashScript -and $cygwinLocation [](start = 10, length = 63)

I think this is:
`if ( test-path $bashScriptInvoker, $bashScript, $cygwinLocation | ?{ ! $_ } ) `

File: test/tools/CodeCoverageAutomation/Start-CodeCoverageRun.ps1 [hunk 3]
```diff
@@ -14,6 +16,30 @@ function Write-LogPassThru
...
+    if not exist "%_CYGBIN%" echo Couldn't find Cygwin at "%_CYGBIN%" & exit 3
+
+    :: Resolve ___.sh to /cygdrive based *nix path and store in %_CYGSCRIPT%
+    for /f "delims=" %%A in ('%_CYGBIN%\cygpath.exe "%~dpn0.sh"') do set _CYGSCRIPT=%%A
+
+    :: Throw away temporary env vars and invoke script, passing any args that were passed to us
+    endlocal & %_CYGBIN%\bash --login "%_CYGSCRIPT%" %*
+'@
+
+    $scriptContent | Out-File $path -Force
```
Comment (JamesWTruher) [inline test/tools/CodeCoverageAutomation/Start-CodeCoverageRun.ps1]: >t-File $path -Force [](start = 23, length = 19)

does this need an explicit encoding?

File: test/tools/CodeCoverageAutomation/Start-CodeCoverageRun.ps1 (hunk 2 above)
Comment (adityapatwardhan) [inline test/tools/CodeCoverageAutomation/Start-CodeCoverageRun.ps1]: The suggested script works for finding if one of them is false, if all is OK it returns null. Instead fixed by using the following:

```
if((Test-Path $bashScriptInvoker) -and
    (Test-Path $bashScript) -and
    (Test-Path $cygwinLocation)
```

File: test/tools/CodeCoverageAutomation/Start-CodeCoverageRun.ps1 (hunk 3 above)
Comment (adityapatwardhan) [inline test/tools/CodeCoverageAutomation/Start-CodeCoverageRun.ps1]: Does not harm.. i will add encoding ascii

Comment (JamesWTruher) [inline test/tools/CodeCoverageAutomation/Start-CodeCoverageRun.ps1]: if it's not needed, it's fine

---
In reply to: [97681882](https://github.com/PowerShell/PowerShell/pull/3037#discussion_r97681882) [](ancestors = 97681882)

**Comment by JamesWTruher** [comment]
:shipit:

**Comment by adityapatwardhan** [comment]
@PowerShell/powershell-maintainers I believe this is ready for merge. Can someone have a look?

**Pull request merged**